4/typedefs.js

```diff
@@ -0,0 +1,214 @@
+      const todoState = args.state || 'UNDONE';
+
+      const session = context.driver.session();
+      try {
+        const cypherCreation = 'CREATE (t:Todo { id: $todoId, message: $message, state: $state })';
+
+        await session.run(cypherCreation, {
+          todoId: todoId,
+          message: args.message,
+          state: todoState
+        });
+        return true;
+      }
+      finally {
+        session.close();
+      }
+    },
+    createAssignee: async (object, args, context, resolveInfo) => {
+      const assigneeId = args.id || uuid();
+
+      const session = context.driver.session();
+      try {
+        const cypherCreation = 'CREATE (a:Assignee { id: $assigneeId, name: $name })';
+
+        await session.run(cypherCreation, {
+          assigneeId: assigneeId,
+          name: args.name
+        });
+        return true;
+      }
+      finally {
+        session.close();
+      }
+    },
+    createRelationship: async (object, args, context, resolveInfo) => {
+      const session = context.driver.session();
+
+      try {
+        const cypherRelation = 'MATCH (a:Assignee), (t:Todo) ' +
+                                'WHERE a.name = $name AND t.id = $id ' +
+                                'CREATE (t)-[r:ASSIGNED_TO]->(a) RETURN type(r)';
+
+        await session.run(cypherRelation, {
+          name: args.name,
+          id: args.id
+        });
+        return true;
+      } finally {
+        session.close();
+      }
+    },
+    deleteTodo: async (object, args, context, resolveInfo) => {
+      const session = context.driver.session();
+      try {
+        const cypherDelete = 'MATCH (t:Todo) WHERE t.id = $id DELETE t';
+        await session.run(cypherDelete, { id: args.id });
+        return true;
+      } finally {
+        session.close();
+      }
+    },
+    deleteAllTodos: async (object, args, context, resolveInfo) => {
+      const session = context.driver.session();
+      try {
+        const cypherDelete = 'MATCH (t:Todo) DETACH DELETE t';
+        await session.run(cypherDelete);
+        return true;
+      } finally {
+        session.close();
+      }
+    },
+    deleteAllAssignees: async (object, args, context, resolveInfo) => {
+      const session = context.driver.session();
+      try {
+        const cypherDelete = 'MATCH (a:Assignee) DETACH DELETE a';
+        await session.run(cypherDelete);
+      } finally {
+        session.close();
+      }
+    },
+    setTodoState: async (object, args, context, resolveInfo) => {
+      const session = context.driver.session();
+      try {
+        const cypherMerge = 'MERGE (t:Todo {id: $id}) ON MATCH SET t.state = $state RETURN t';
+        const result = await session.run(cypherMerge, {
+          id: args.id,
+          state: args.state
+        });
+        const todo = result.records.map(record => record.get('t').properties)[0];
+        return todo;
+      } finally {
+        session.close();
+      }
+    }
+  },
+};
```

Comment:
Maybe export resolvers to individual file to enhance the code overview (not obligatory tough)